File: suc/swift/030.swift
import SwiftUI

// MARK: - AppTokens
// 集中管理所有设计令牌，方便原子级修改和保持双平台一致性。
struct AppTokens {
    struct Colors {
        // 使用 hex 初始化 Color，确保颜色精确对应
        static let primary = Color(hex: 0xFF007BFF)
        static let secondary = Color(hex: 0xFF6C757D)
        static let tertiary = Color(hex: 0xFF17A2B8)
        static let background = Color(hex: 0xFFF8F9FA)
        static let surface = Color(hex: 0xFFFFFFFF)
        static let surfaceVariant = Color(hex: 0xFFE9ECEF)
        static let outline = Color(hex: 0xFFDEE2E6)
        static let success = Color(hex: 0xFF28A745)
        static let warning = Color(hex: 0xFFFFC107)
        static let error = Color(hex: 0xFFDC3545)
        static let onPrimary = Color(hex: 0xFFFFFFFF)
        static let onSecondary = Color(hex: 0xFFFFFFFF)
        static let onTertiary = Color(hex: 0xFFFFFFFF)
        static let onBackground = Color(hex: 0xFF212529)
        static let onSurface = Color(hex: 0xFF212529)
    }

    struct TypographyTokens {
        // 字体大小和字重与 Compose 对应
        static let display = Font.system(size: 36, weight: .bold)
        static let headline = Font.system(size: 28, weight: .semibold)
        static let title = Font.system(size: 22, weight: .medium)
        static let body = Font.system(size: 16, weight: .regular) // Compose 的 FontWeight.Normal 对应 Swift 的 .regular
        static let label = Font.system(size: 14, weight: .medium)
    }

    struct Shapes {
        // 圆角半径
        static let small: CGFloat = 4
        static let medium: CGFloat = 8
        static let large: CGFloat = 12
    }

    struct Spacing {
        // 间距值，直接从 Dp 转换为 CGFloat
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 12
        static let lg: CGFloat = 16
        static let xl: CGFloat = 24
    }

    // 阴影规范，用于精确匹配 Compose 的 elevation
    struct ShadowSpec {
        let elevation: CGFloat // 对应 Compose 的 elevation，用于计算阴影强度
        let radius: CGFloat    // 阴影模糊半径
        let dy: CGFloat        // Y 轴偏移
        let opacity: Double    // 阴影不透明度
    }

    struct ElevationMapping {
        static let level1 = ShadowSpec(elevation: 1, radius: 3, dy: 1, opacity: 0.1)
        static let level2 = ShadowSpec(elevation: 3, radius: 6, dy: 2, opacity: 0.1)
        static let level3 = ShadowSpec(elevation: 6, radius: 10, dy: 4, opacity: 0.1)
    }
}

// 扩展 Color，支持从十六进制值初始化
extension Color {
    init(hex: UInt, alpha: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xff) / 255,
            green: Double((hex >> 08) & 0xff) / 255,
            blue: Double((hex >> 00) & 0xff) / 255,
            opacity: alpha
        )
    }
}

// MARK: - App Theme Environment Values
// 使用 EnvironmentKey 模拟 Compose 的 MaterialTheme.typography 访问方式
struct AppTypographyKey: EnvironmentKey {
    static let defaultValue: AppTypography = AppTypography()
}

struct AppTypography {
    let displayMedium = AppTokens.TypographyTokens.display
    let headlineSmall = AppTokens.TypographyTokens.headline
    let titleMedium = AppTokens.TypographyTokens.title
    let bodyMedium = AppTokens.TypographyTokens.body
    let labelMedium = AppTokens.TypographyTokens.label
}

extension EnvironmentValues {
    var appTypography: AppTypography {
        get { self[AppTypographyKey.self] }
        set { self[AppTypographyKey.self] = newValue }
    }
}

// 安全区域 Environment Key
struct SafeAreaInsetsKey: EnvironmentKey {
    static var defaultValue: EdgeInsets {
        EdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 0)
    }
}

extension EnvironmentValues {
    var safeAreaInsets: EdgeInsets {
        get { self[SafeAreaInsetsKey.self] }
        set { self[SafeAreaInsetsKey.self] = newValue }
    }
}

// MARK: - Data Models
// 对应 Kotlin 的 ClipboardItem 数据类
struct ClipboardItem: Identifiable {
    let id: Int
    let content: String
    let type: String
    let timestamp: String
    let progress: Float
}

// MARK: - RootScreen
// 对应 Kotlin 的 RootScreen Composable
struct RootScreen: View {
    @State private var sliderValue: Float = 24.0

    // 模拟 Compose 中的 remember { listOf(...) }
    let clipboardItems: [ClipboardItem] = [
        ClipboardItem(id: 1, content: "https://www.example.com/modern-design", type: "Link", timestamp: "5 min ago", progress: 0.2),
        ClipboardItem(id: 2, content: "Final project review notes have been updated.", type: "Text", timestamp: "23 min ago", progress: 0.5),
        ClipboardItem(id: 3, content: "#007BFF", type: "Color", timestamp: "1 hour ago", progress: 0.8),
        ClipboardItem(id: 4, content: "Meeting at 3 PM with the design team.", type: "Text", timestamp: "3 hours ago", progress: 0.9),
        ClipboardItem(id: 5, content: "Shared file: 'Q3_Report.pdf'", type: "File", timestamp: "Yesterday", progress: 1.0)
    ]

    var body: some View {
        GeometryReader { geometry in
            // 整体使用 VStack 布局，模拟 Scaffold
            VStack(spacing: 0) {
                // 自定义顶部栏，模拟 CenterAlignedTopAppBar
                HStack {
                    Spacer() // 用于居中标题
                    Text("Clipboard History")
                        .font(AppTokens.TypographyTokens.title)
                        .foregroundColor(AppTokens.Colors.onBackground)
                    Spacer() // 用于居中标题

                    Button(action: {
                        // 处理 Clear 按钮点击事件
                        print("Clear button tapped")
                    }) {
                        Text("Clear")
                            .font(AppTokens.TypographyTokens.label)
                            .foregroundColor(AppTokens.Colors.onSurface)
                            .padding(.vertical, AppTokens.Spacing.sm)
                            .padding(.horizontal, AppTokens.Spacing.md)
                            .background(AppTokens.Colors.surfaceVariant)
                            .cornerRadius(AppTokens.Shapes.medium)
                    }
                    .padding(.trailing, AppTokens.Spacing.md) // 按钮右侧间距
                }
                .padding(.top, geometry.safeAreaInsets.top) // 顶部内边距，避开状态栏
                .frame(height: 56 + geometry.safeAreaInsets.top) // 顶部栏高度 (标准 AppBar 高度 + 状态栏高度)
                .background(AppTokens.Colors.surface)
                // 顶部栏阴影，模拟 elevation
                .shadow(color: .black.opacity(0.05), radius: 0.5, x: 0, y: 0.5)

                // 内容区域，模拟 LazyColumn
                ScrollView {
                    LazyVStack(spacing: AppTokens.Spacing.md) { // 垂直间距
                        ForEach(clipboardItems) { item in
                            ClipboardCard(item: item)
                        }
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg) // 水平内边距
                    .padding(.vertical, AppTokens.Spacing.md)   // 垂直内边距
                }
                .background(AppTokens.Colors.background) // 滚动区域背景色

                // 设置面板
                SettingsPane(sliderValue: $sliderValue)
                    .background(AppTokens.Colors.surface)
                    // 设置面板阴影，模拟 elevation
                    .shadow(color: .black.opacity(AppTokens.ElevationMapping.level2.opacity),
                            radius: AppTokens.ElevationMapping.level2.radius,
                            x: 0,
                            y: -AppTokens.ElevationMapping.level2.dy) // 阴影向上
            }
        }
        // 忽略安全区域，使内容延伸到屏幕边缘，包括状态栏下方
        .ignoresSafeArea(.all, edges: .top)
        .background(AppTokens.Colors.background) // 整个屏幕的背景色
        .statusBarHidden(true) // 隐藏状态栏
    }
}

// MARK: - ClipboardCard
// 对应 Kotlin 的 ClipboardCard Composable
struct ClipboardCard: View {
    let item: ClipboardItem
    // 通过 Environment 访问自定义的排版样式
    @Environment(\.appTypography) var typography

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            HStack(alignment: .center, spacing: AppTokens.Spacing.sm) {
                // 模拟 Box + clip + background
                RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                    .fill(AppTokens.Colors.primary)
                    .frame(width: 24, height: 24)

                Text(item.type)
                    .font(typography.labelMedium)
                    .foregroundColor(AppTokens.Colors.primary)
                    .fontWeight(.bold)

                Spacer() // 模拟 Modifier.weight(1f)

                Text(item.timestamp)
                    .font(typography.labelMedium)
                    .foregroundColor(AppTokens.Colors.secondary)
            }
            .padding(.bottom, AppTokens.Spacing.sm) // 模拟 Spacer height

            Text(item.content)
                .font(typography.bodyMedium)
                .foregroundColor(AppTokens.Colors.onSurface)
                .lineLimit(2) // 模拟 maxLines
                .truncationMode(.tail) // 模拟 TextOverflow.Ellipsis
                .padding(.bottom, AppTokens.Spacing.md) // 模拟 Spacer height

            // 线性进度指示器，使用自定义样式以匹配 Compose 的 trackColor
            ProgressView(value: item.progress)
                .progressViewStyle(CustomLinearProgressViewStyle(tint: AppTokens.Colors.tertiary, track: AppTokens.Colors.surfaceVariant))
                .frame(height: 4) // 设置高度
        }
        .padding(AppTokens.Spacing.md) // 内边距
        .background(AppTokens.Colors.surface) // 背景色
        .cornerRadius(AppTokens.Shapes.large) // 圆角
        .overlay(
            // 边框，模拟 BorderStroke
            RoundedRectangle(cornerRadius: AppTokens.Shapes.large)
                .stroke(AppTokens.Colors.surfaceVariant, lineWidth: 1)
        )
        // 阴影，模拟 elevation
        .shadow(color: .black.opacity(AppTokens.ElevationMapping.level1.opacity),
                radius: AppTokens.ElevationMapping.level1.radius,
                x: 0,
                y: AppTokens.ElevationMapping.level1.dy)
    }
}

// 自定义 ProgressViewStyle，以匹配 Compose 的 LinearProgressIndicator 的 trackColor
struct CustomLinearProgressViewStyle: ProgressViewStyle {
    var tint: Color // 进度条颜色
    var track: Color // 轨道颜色

    func makeBody(configuration: Configuration) -> some View {
        GeometryReader { geometry in
            ZStack(alignment: .leading) {
                // 轨道背景
                RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                    .fill(track)
                    .frame(height: 4)
                    .frame(width: geometry.size.width) // 轨道宽度填充可用空间

                // 进度条
                RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                    .fill(tint)
                    .frame(width: CGFloat(configuration.fractionCompleted ?? 0) * geometry.size.width, height: 4)
            }
        }
        .frame(height: 4) // 确保整个 ProgressView 的高度
    }
}

// MARK: - SettingsPane
// 对应 Kotlin 的 SettingsPane Composable
struct SettingsPane: View {
    @Binding var sliderValue: Float // 绑定滑块值
    @Environment(\.appTypography) var typography

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            HStack {
                Text("History Limit (Hours)")
                    .font(typography.labelMedium)
                    .foregroundColor(AppTokens.Colors.onSurface)
                Spacer()
                Text(String(format: "%.0f", sliderValue)) // 显示整数值
                    .font(typography.bodyMedium)
                    .fontWeight(.semibold)
                    .foregroundColor(AppTokens.Colors.onSurface)
            }
            .padding(.bottom, AppTokens.Spacing.sm) // 垂直间距

            Slider(value: $sliderValue, in: 1...72, step: 1) 
                .tint(AppTokens.Colors.primary) // iOS 15+
                .accentColor(AppTokens.Colors.primary) // 兼容 iOS 14
        }
        .padding(AppTokens.Spacing.lg) // 内边距
        .frame(maxWidth: .infinity) // 宽度填充
        // 面板的背景和阴影在 RootScreen 中应用
    }
}

// MARK: - App Entry Point
// 对应 Kotlin 的 MainActivity，使用 @main 标记为应用入口
@main
struct ClipboardHistoryApp: App {
    var body: some Scene {
        WindowGroup {
            RootScreen()
                // 将自定义的排版样式注入环境，供子视图使用
                .environment(\.appTypography, AppTypography())
        }
    }
}

// MARK: - Preview
// Xcode 预览，对应 Compose 的 @Preview
struct RootScreen_Previews: PreviewProvider {
    static var previews: some View {
        RootScreen()
            .environment(\.appTypography, AppTypography())
    }
}

File: suc/kotlin/030.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Slider
import androidx.compose.material3.SliderDefaults
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "031_ClipboardHistory_en"
private const val UI_TYPE = "Productivity"
private const val STYLE_THEME = "Modern Blue"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF007BFF)
        val secondary = Color(0xFF6C757D)
        val tertiary = Color(0xFF17A2B8)
        val background = Color(0xFFF8F9FA)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFE9ECEF)
        val outline = Color(0xFFDEE2E6)
        val success = Color(0xFF28A745)
        val warning = Color(0xFFFFC107)
        val error = Color(0xFFDC3545)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF212529)
        val onSurface = Color(0xFF212529)
    }

    object TypographyTokens {
        val display = TextStyle(fontSize = 36.sp, fontWeight = FontWeight.Bold)
        val headline = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.SemiBold)
        val title = TextStyle(fontSize = 22.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Normal, lineHeight = 24.sp)
        val label = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.25.sp)
    }

    object Shapes {
        val small = RoundedCornerShape(4.dp)
        val medium = RoundedCornerShape(8.dp)
        val large = RoundedCornerShape(12.dp)
    }

    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 12.dp
        val lg = 16.dp
        val xl = 24.dp
    }

    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 3.dp, 1.dp, 0.1f)
        val level2 = ShadowSpec(3.dp, 6.dp, 2.dp, 0.1f)
        val level3 = ShadowSpec(6.dp, 10.dp, 4.dp, 0.1f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayMedium = AppTokens.TypographyTokens.display,
    headlineSmall = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(small = AppTokens.Shapes.small, medium = AppTokens.Shapes.medium, large = AppTokens.Shapes.large),
        content = content
    )
}

data class ClipboardItem(val id: Int, val content: String, val type: String, val timestamp: String, val progress: Float)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    val clipboardItems = remember {
        listOf(
            ClipboardItem(1, "https://www.example.com/modern-design", "Link", "5 min ago", 0.2f),
            ClipboardItem(2, "Final project review notes have been updated.", "Text", "23 min ago", 0.5f),
            ClipboardItem(3, "#007BFF", "Color", "1 hour ago", 0.8f),
            ClipboardItem(4, "Meeting at 3 PM with the design team.", "Text", "3 hours ago", 0.9f),
            ClipboardItem(5, "Shared file: 'Q3_Report.pdf'", "File", "Yesterday", 1.0f)
        )
    }
    var sliderValue by remember { mutableStateOf(24f) }

    Scaffold(
        contentWindowInsets = WindowInsets(0),
        containerColor = AppTokens.Colors.background,
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("Clipboard History", style = MaterialTheme.typography.titleMedium) },
                actions = {
                    Button(
                        onClick = {},
                        colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.surfaceVariant, contentColor = AppTokens.Colors.onSurface),
                        shape = AppTokens.Shapes.medium,
                        modifier = Modifier.padding(end = AppTokens.Spacing.md)
                    ) {
                        Text("Clear", style = MaterialTheme.typography.labelMedium)
                    }
                }
            )
        }
    ) { paddingValues ->
        Column(modifier = Modifier.fillMaxSize().padding(paddingValues)) {
            LazyColumn(
                modifier = Modifier.weight(1f),
                contentPadding = PaddingValues(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md),
                verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
            ) {
                items(clipboardItems) { item ->
                    ClipboardCard(item)
                }
            }
            SettingsPane(sliderValue = sliderValue, onSliderChange = { sliderValue = it })
        }
    }
}

@Composable
fun ClipboardCard(item: ClipboardItem) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = AppTokens.Shapes.large,
        colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level1.elevation),
        border = BorderStroke(1.dp, AppTokens.Colors.surfaceVariant)
    ) {
        Column(modifier = Modifier.padding(AppTokens.Spacing.md)) {
            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)) {
                Box(modifier = Modifier.size(24.dp).clip(AppTokens.Shapes.small).background(AppTokens.Colors.primary)) {}
                Text(item.type, style = MaterialTheme.typography.labelMedium, color = AppTokens.Colors.primary, fontWeight = FontWeight.Bold)
                Spacer(modifier = Modifier.weight(1f))
                Text(item.timestamp, style = MaterialTheme.typography.labelMedium, color = AppTokens.Colors.secondary)
            }
            Spacer(modifier = Modifier.height(AppTokens.Spacing.sm))
            Text(item.content, style = MaterialTheme.typography.bodyMedium, maxLines = 2, overflow = TextOverflow.Ellipsis)
            Spacer(modifier = Modifier.height(AppTokens.Spacing.md))
            LinearProgressIndicator(
                progress = { item.progress },
                modifier = Modifier.fillMaxWidth().height(4.dp).clip(AppTokens.Shapes.small),
                color = AppTokens.Colors.tertiary,
                trackColor = AppTokens.Colors.surfaceVariant
            )
        }
    }
}

@Composable
fun SettingsPane(sliderValue: Float, onSliderChange: (Float) -> Unit) {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        color = AppTokens.Colors.surface,
        tonalElevation = AppTokens.ElevationMapping.level2.elevation,
        shadowElevation = AppTokens.ElevationMapping.level2.elevation
    ) {
        Column(modifier = Modifier.padding(AppTokens.Spacing.lg)) {
            Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
                Text("History Limit (Hours)", style = MaterialTheme.typography.labelMedium, color = AppTokens.Colors.onSurface)
                Spacer(modifier = Modifier.weight(1f))
                Text(sliderValue.toInt().toString(), style = MaterialTheme.typography.bodyMedium, fontWeight = FontWeight.SemiBold)
            }
            Slider(
                value = sliderValue,
                onValueChange = onSliderChange,
                valueRange = 1f..72f,
                steps = 70,
                modifier = Modifier.fillMaxWidth(),
                colors = SliderDefaults.colors(
                    thumbColor = AppTokens.Colors.primary,
                    activeTrackColor = AppTokens.Colors.primary,
                    inactiveTrackColor = AppTokens.Colors.surfaceVariant,
                    activeTickColor = AppTokens.Colors.primary.copy(alpha = 0.5f),
                    inactiveTickColor = AppTokens.Colors.surfaceVariant
                )
            )
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, widthDp = 360, heightDp = 720)
@Composable
fun DefaultPreview() {
    AppTheme {
        RootScreen()
    }
}
